scrapy_project/spiders/book_spider.py

```diff
@@ -0,0 +1,59 @@
+import re
+
+import scrapy
+from scrapy.http import Response
+
+
+class BookSpiderSpider(scrapy.Spider):
+    name = "book_spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> scrapy.Request:
+        for book in response.css(".product_pod"):
+            detail_url = book.css("h3 a::attr(href)").get()
+            detail_url = response.urljoin(detail_url)
+
+            yield scrapy.Request(
+                detail_url,
+                callback=self.parse_book_details
+            )
+
+        next_page = response.css(".next a::attr(href)").get()
+        if next_page:
+            next_page = response.urljoin(next_page)
+            yield scrapy.Request(
+                next_page,
+                callback=self.parse
+            )
+
+    def parse_book_details(self, response: Response) -> dict:
```

Comment:
The return type annotation for the `parse_book_details` method should be `Generator` or `Iterable` instead of `dict`, as this method yields dictionaries.